src/controller/CHIPDeviceControllerFactory.h

```diff
@@ -106,6 +106,13 @@ class DeviceControllerFactory
     DeviceControllerFactory(DeviceControllerFactory const &) = delete;
     void operator=(DeviceControllerFactory const &) = delete;
 
+    //
+    // Some clients do not prefer a complete shutdown of the stack being initiated if
+    // all device controllers have ceased to exist. To avoid that, this method has been
+    // created to permit retention of the underlying system state to avoid that.
```

Comment:
Who will release it and when?

Reply:
It'll be released at stack shutdown in the REPL. Have updated the comments and added a release method.